test/unittests/stack_test.cpp

```diff
@@ -3,130 +3,118 @@
+    EXPECT_EQ(stack[1], 12);
+    EXPECT_EQ(stack[2], 11);
 
-    stack.drop(0);
-    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.pop(), 13);
     EXPECT_EQ(stack.size(), 2);
-
-    stack.clear();
-    EXPECT_TRUE(stack.empty());
-    EXPECT_EQ(stack.size(), 0);
+    EXPECT_EQ(stack.top(), 12);
 }
 
-TEST(stack, resize)
+TEST(operand_stack, large)
 {
-    Stack<char> stack;
-    stack.push('a');
-    stack.push('b');
-
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 2);
-
-    // grow stack
-    stack.resize(4);
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.peek(), 0);
-    EXPECT_EQ(stack.peek(1), 0);
-    EXPECT_EQ(stack.peek(2), 'b');
-    EXPECT_EQ(stack.peek(3), 'a');
-    EXPECT_EQ(stack.size(), 4);
-
-    stack.drop();
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 3);
+    constexpr auto max_height = 33;
+    OperandStack stack(max_height);
 
-    // shrink stack
-    stack.resize(1);
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 1);
+    for (unsigned i = 0; i < max_height; ++i)
+        stack.push(i);
 
-    stack.clear();
-    EXPECT_TRUE(stack.empty());
+    EXPECT_EQ(stack.size(), max_height);
+    for (int expected = max_height - 1; expected >= 0; --expected)
+        EXPECT_EQ(stack.pop(), expected);
+    EXPECT_EQ(stack.size(), 0);
 }
 
-TEST(stack, iterator)
+TEST(operand_stack, shrink)
 {
-    Stack<char> stack;
-    stack.push('a');
-    stack.push('b');
-    stack.push('c');
-
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 3);
-
-    auto bottom_item = stack.begin();
-    EXPECT_EQ(*bottom_item, 'a');
-
-    auto top_item = stack.end() - 1;
-    EXPECT_EQ(*top_item, 'c');
-
-    for (auto it = stack.begin(); it < stack.end(); it++)
-        *it = 'x';
+    constexpr auto max_height = 60;
+    OperandStack stack(max_height);
+
+    for (unsigned i = 0; i < max_height; ++i)
+        stack.push(i);
+
+    EXPECT_EQ(stack.size(), max_height);
+    constexpr auto new_height = max_height / 3;
+    stack.shrink(new_height);
+    EXPECT_EQ(stack.size(), new_height);
+    EXPECT_EQ(stack.top(), new_height - 1);
+    EXPECT_EQ(stack[0], new_height - 1);
+    EXPECT_EQ(stack[new_height - 1], 0);
+}
 
-    for (unsigned item = 0; item < stack.size(); item++)
-        EXPECT_EQ(stack.peek(item), 'x');
+TEST(operand_stack, rbegin_rend)
+{
+    OperandStack stack(3);
+    EXPECT_EQ(stack.rbegin(), stack.rend());
+
+    stack.push(1);
+    stack.push(2);
+    stack.push(3);
+    EXPECT_NE(stack.rbegin(), stack.rend());
+    EXPECT_EQ(*stack.rbegin(), 1);
+    EXPECT_EQ(*(stack.rend() - 1), 3);
 }
 
-TEST(stack, clear_on_empty)
+TEST(operand_stack, to_vector)
 {
-    Stack<char> stack;
-    stack.clear();
+    OperandStack stack(3);
+    EXPECT_TRUE(std::vector(stack.rbegin(), stack.rend()).empty());
+
+    stack.push(1);
+    stack.push(2);
+    stack.push(3);
+
+    EXPECT_EQ(std::vector(stack.rbegin(), stack.rend()), (std::vector<uint64_t>{1, 2, 3}));
```

Comment:
Maybe `EXPECT_THAT` would be nice here
```suggestion
    EXPECT_THAT(std::vector(stack.rbegin(), stack.rend()), ElementsAre(1, 2, 3));
```